bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -105,49 +119,78 @@ pub mod pallet {
 		/// Convert versioned location failure
 		UnsupportedLocationVersion,
 		/// Check location failure, should start from the dispatch origin as owner
-		OwnerCheck,
+		InvalidAssetOwner,
 		/// Send xcm message failure
 		Send,
 		/// Withdraw fee asset failure
 		FundsUnavailable,
+		/// Convert to reanchored location failure
+		LocationConversionFailed,
 	}
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 		/// Call `create_agent` to instantiate a new agent contract representing `origin`.
-		/// - `origin`: Can be any sovereign `Location`
+		/// - `origin`: `Location` can be from either of:
+		/// 	a. any kind of location from a sibling parachain
+		/// 	b. sudo from local, location will be `Here` in this case.
+		/// 	c. a signed origin from local, location will be `(0,[AccountId32()])` in this case.
 		/// - `fee`: Fee in Ether paying for the execution cost on Ethreum
 		#[pallet::call_index(1)]
 		#[pallet::weight(T::WeightInfo::create_agent())]
 		pub fn create_agent(origin: OriginFor<T>, fee: u128) -> DispatchResult {
 			let origin_location = T::CreateAgentOrigin::ensure_origin(origin)?;
 
-			Self::burn_fees(origin_location.clone(), fee)?;
+			// Ignore fee charges for sudo call
+			if !origin_location.eq(&Here.into()) {
+				// Burn Ether Fee for the cost on ethereum
+				Self::burn_for_teleport(&origin_location, &(T::FeeAsset::get(), fee).into())?;
+
+				// Burn RemoteExecutionFee for the cost on bridge hub
+				Self::burn_for_teleport(&origin_location, &T::RemoteExecutionFee::get())?;
+			}
 
-			let call = SnowbridgeControl::Control(ControlCall::CreateAgent {
-				location: Box::new(VersionedLocation::from(origin_location.clone())),
+			let reanchored_location = origin_location
```

Comment:
I feel like BH should take responsibility for reanchoring. Does it not do that already?